pkg/clusteragent/admission/mutate/autoinstrumentation/tag_resolver.go

```diff
@@ -0,0 +1,177 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoinstrumentation
+
+import (
+	"encoding/json"
+	"sync"
+
+	rcclient "github.com/DataDog/datadog-agent/pkg/config/remote/client"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// TagResolver is a component that resolves image tags from a remote configuration.
```

Comment:
Per another comment, I think it would be good to have an interface and have an implementation for a `noopTagResolver` that always returns the tag directly, and another that has one for an `rcclient` if its there.

The other thing I'm seeing here is that sometimes you're doing `nil` checks, but `NewTagResolver` never returns `nil`.